src/consensus/proposer.rs

```diff
@@ -19,7 +19,37 @@ use tokio::time::Instant;
 use tokio::{select, time};
 use tracing::{error, warn};
 
-pub const PROTOCOL_VERSION: u32 = 1;
+use super::validator::ProposalSource;
+
+#[derive(Eq, PartialEq, Clone, Copy)]
+pub enum ProtocolVersion {
+    UNKNOWN = 0,
+    ONE = 1,
+    TWO = 2,
+}
+
+struct VersionSchedule {
+    version: ProtocolVersion,
+    active_at: u64,
```

Comment:
Worth adding a comment about what kind of timestamp this is (seconds since unix epoch?)